DICOM/src/Microsoft.Health.DeID.SharedLib/RedactFunction.cs

```diff
@@ -0,0 +1,166 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Linq;
+using System.Text.RegularExpressions;
+using EnsureThat;
+using Microsoft.Health.DeID.SharedLib.Models;
+using Microsoft.Health.Dicom.DeID.SharedLib.Models;
+using Microsoft.Health.Dicom.DeID.SharedLib.Settings;
+
+namespace Microsoft.Health.Dicom.DeID.SharedLib
+{
+    public class RedactFunction
+    {
+        private static readonly int AgeThreshold = 89;
+        private static readonly string ReplacementDigit = "0";
+        private static readonly int InitialDigitsCount = 3;
+        private readonly RedactSetting _redactSetting;
+
+        public RedactFunction(RedactSetting redactSetting = null)
+        {
+            _redactSetting = redactSetting ?? new RedactSetting();
+        }
+
+        public string Redact(string inputString, AnonymizerValueTypes valueType)
+        {
+            return valueType switch
+            {
+                AnonymizerValueTypes.Date => RedactDate(inputString),
+                AnonymizerValueTypes.DateTime => RedactDateTime(inputString),
+                AnonymizerValueTypes.Age => RedactAge(decimal.Parse(inputString)).ToString(),
```

Comment:
❔ [Question]
This may throw an exception here, and you don't need to process it, right?

Reply:
Add exception handle.